src/cmap/conn/command.rs

```diff
@@ -14,37 +13,26 @@ use crate::{
     ClientSession,
 };
 
-/// A command that has been serialized to BSON.
-#[derive(Debug)]
-pub(crate) struct RawCommand {
-    pub(crate) name: String,
-    pub(crate) target_db: String,
-    /// Whether or not the server may respond to this command multiple times via the moreToComeBit.
-    pub(crate) exhaust_allowed: bool,
-    pub(crate) bytes: Vec<u8>,
-}
-
-impl RawCommand {
-    pub(crate) fn should_compress(&self) -> bool {
-        let name = self.name.to_lowercase();
-        !REDACTED_COMMANDS.contains(name.as_str()) && !HELLO_COMMAND_NAMES.contains(name.as_str())
-    }
-}
-
 /// Driver-side model of a database command.
 #[serde_with::skip_serializing_none]
 #[derive(Clone, Debug, Serialize, Default)]
 #[serde(rename_all = "camelCase")]
-pub(crate) struct Command<T = Document> {
+pub(crate) struct Command<T = Document>
+where
+    T: Serialize,
+{
     #[serde(skip)]
     pub(crate) name: String,
 
     #[serde(skip)]
     pub(crate) exhaust_allowed: bool,
 
-    #[serde(flatten)]
+    #[serde(skip)]
```

Comment:
AFAICT the new behavior (`skip` and `extend_raw_document_buf` in `Message::from_command`) should be equivalent to `flatten`?

Reply:
Oh yeah you're right, reverted